docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,223 @@
+
+## Document Purpose
+
+Propose a solution to reliably join mobile session data between daemon and main application telemetry.
+
+## Requirements
+
+The main transport for session related data are the session pings, namely
+the [`vpnsession`](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/pings/vpnsession) ping
+and the [`daemonsession`](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/pings/daemonsession) ping.
+
+It is a requirement on the proposed solution, that the metrics included in each ping are all
+recorded during the session related to the shared session id in it i.e. the ping does not contain
+any metric from a previous session.
+
+## Proposed Solution
+
+A new metric will be introduced to mobile platforms: `shared_session_id`. This metric will be the
+primary session identifier metric for Mozilla VPN mobile telemetry. This metric will have lifetime
+`User`, with this lifetime, once a metric is set it can only be overwritten but never cleared from
+the telemetry storage.
+
+The daemon process will be responsible for managing this metric. The daemon is the preferred
+place for this, because it is _guaranteed_ to be running throughout the whole VPN session.
+
+The main application will communicate with the daemon in order to get the correct value for
+this metric whenever necessary.
+
+The scheduling of the `vpnsession` and `daemonsession` pings will be tweaked to meet
+the requirements of isolating session data per ping.
+
+### On the daemon
+
+There are three important triggers that require action on the shared session id and
+the `daemonsession` ping on the daemon process.
+
+#### On session start
+
+Once a new session is started,
+
+1. The `daemonsession` ping will be submitted with reason `flush`, in order to flush out
+   any lingering data before starting collection telemetry related to the new session.
+2. The `shared_session_id` is generated and recorded.
+3. The value recorded to telemetry is also persisted on the daemon's local storage.
+4. Session metrics may be recorded from this point on.
+5. Session is started.
+
+The `flush` ping submitted on step one is expected to be empty. The submission of it right
+before the start of a new session is a safety measure to guarantee no lingering data makes it
+to the actual session ping.
+
+#### On session end
+
+Once an ongoing session is ended,
+
+1. The `daemonsession` ping is submitted with the reason `daemon_end`.
+2. The daemon's local storage for the `shared_session_id` is cleared.
+3. The `shared_session_id` metric is set to a known value. Due to its `User` lifetime,
+   this metric cannot be cleared from storage ever. To indicate a period of inactivity,
+   we will leverage a known UUID.
+
+#### On daemon process start
+
+The session end trigger may not be hit, if the daemon process is force killed. Therefore,
+whenever the daemon process is started the local storage will be searched for a lingering
```

Comment:
We don't mean lingering anymore, right? We want "known inactive session ID"?

Reply:
It is lingering in this case. The daemon is still managing the UUID in it's own storage. A lingering session id for the daemon means the end event was not executed succesfully and there may be lingering data.

Reply:
Yup, total brainfart when I wrote that comment. Thanks.